src/net/sourceforge/kolmafia/session/MallPriceManager.java

```diff
@@ -415,64 +418,195 @@ public static final void searchMall(
+
+    if (price == 0) {
+      AdventureResult search = ItemPool.get(itemId, NTH_CHEAPEST_PRICE);
+      List<PurchaseRequest> results = MallPriceManager.searchMall(search);
+      MallPriceManager.updateMallPrice(itemId, results);
+      price = MallPriceManager.mallPrices.getOrDefault(itemId, 0);
     }
 
     return price;
   }
 
-  public static final synchronized int getMallPrice(final AdventureResult item) {
+  // Get the up-to-date "nth cheapest" mall price from cached local mall searches.
+  //
+  // The "count" field of the AdventureResult is meaningful; if it is no greater than "n",
+  // we'll use a cached mall price, which is the "nth cheapest" price,
+  // and return the price to buy "count" items.
+  //
+  // If it is greater than "n", and we have a saved mall search which will accommodate that, we'll
+  // calculate the total price using the "nth cheapest" price for the first 5 and the actual price
+  // for additional items.
+  //
+  // If we need to make a mall search to get enough items, we will do so - and save the search and
+  // the calculated "nth cheapest" price for subsequent calls.
+  //
+  // Note that this method returns a "long" value, since the cost to acquire multiple items can
+  // exceed an int.
+
+  public static long getMallPrice(final AdventureResult item) {
+    // Don't waste time if the item is not purchasable.
     int itemId = item.getItemId();
-    if (itemId < 1
-        || (!ItemDatabase.isTradeable(itemId) && !NPCStoreDatabase.contains(itemId, true))) {
+    if (!validMallItem(itemId)) {
       return 0;
     }
-    if (MallPriceManager.mallPrices.getOrDefault(itemId, 0) == 0) {
-      List<PurchaseRequest> results =
-          MallPriceManager.searchMall(item.getInstance(NTH_CHEAPEST_PRICE));
-      MallPriceManager.updateMallPrice(item, results);
+
+    // If we want to know the price of no more than the "n" of an item, use a cached mall price
+    int count = Math.max(1, item.getCount());
+    if (count <= NTH_CHEAPEST_PRICE) {
+      return (long) MallPriceManager.getMallPrice(itemId) * count;
+    }
+
+    // Do a mall search. Any cached result that is not stale is acceptable.
+    List<PurchaseRequest> results = MallPriceManager.searchMall(item.getInstance(0));
+
+    // Iterate through the PurchaseRequests accumulating prices.
+    int needed = count;
+    int found = 0;
+    int price = -1;
+
+    long total = 0;
+
+    for (PurchaseRequest req : results) {
+      price = req.getPrice();
+      int available = req.getLimit();
+
+      // If we are still looking for the "nth-cheapest price" ...
+      if (found < NTH_CHEAPEST_PRICE) {
+        // If this store does not have it, simply account for available inventory
+        if ((found + available) < NTH_CHEAPEST_PRICE) {
+          // Don't add price until we have at least 5
+          found += available;
+          continue;
+        }
+
+        // Otherwise, we have found the 5th-cheapest price!
```

Comment:
```suggestion
        // Otherwise, we have found the nth-cheapest price!
```
nth cheapest vs 5th cheapest. Probably nth cheapest is more consistent.

Reply:
Yes. I clarified all comments. The magic number "5" is now only mentioned in the declaration of the constant.